---
title: "ESP_CTD_direct_comparisons"
author: "Markus Min"
date: "7/27/2021"
output: html_document
---


### Description
In this Rmd file, we are analyzing the direct comparison samples from the CANON17S cruise, which were 10L water samples that were sampled in two ways: through the ESP and via the peristaltic pump. We have six samples, and it appears that these samples were only ever sequenced for COI and 16S. We will run through the COI analyses first, and then the 16S.

### Load libraries
```{r load_libraries}
library(tidyverse)
library(here)
library(readxl)
library(phyloseq)
library(qiime2R)
library(ggthemes)
library(ggpubr)
library(janitor)

fig_dir <- here::here("figures", "direct_comparisons")
```


# COI

### load data
```{r load_data}
# load asv table
directcomp_asv_table <- read.csv(here("data", "COI", "direct_comparisons", "directcomp_ESP_CTD_COI_asv_table.csv"), row.names = 1)

# load tax table
directcomp_tax_table <- read.csv(here("data", "COI", "direct_comparisons", "directcomp_ESP_CTD_COI_tax_table.csv"), row.names = 1)

# load metadata
directcomp_metadata <- read.csv(here("data", "COI", "direct_comparisons", "directcomp_ESP_CTD_COI_metadata.csv"), row.names = 1)

### Merge into phyloseq
directcomp_COI_phyloseq <- merge_phyloseq(otu_table(directcomp_asv_table,taxa_are_rows = TRUE),
                                       tax_table(as.matrix(directcomp_tax_table)),
                                       sample_data(directcomp_metadata))


### Load other metadata (Kevan Yamahara's corrected data)
directcomp_meta_yamahara <- read_excel(here("data", "ESP_CTD_matched_samples_metadata_YAMAHARACORRECTED.xlsx"), sheet = "CN17S")

# Add another column to show which samples are paired
directcomp_meta_yamahara %>% 
  mutate(., sampling_pair = ifelse(sample_name %in% c("CANON_B1_22", "CANON_CBI1_22"), "1",
                                   ifelse(sample_name %in% c("CANON_B10_22", "CANON_CBI4_22"), "2",
                                          ifelse(sample_name %in% c("CANON_B13_22", "CANON_CBI5_22"), "3",
                                                 ifelse(sample_name %in% c("CANON_B17_22", "CANON_CBI8_22"), "4",
                                                        ifelse(sample_name %in% c("CANON_B20_22", "CANON_CBI9_22"), "5",
                                                               ifelse(sample_name %in% c("CANON_B23_22", "CANON_CBI10_22"), "6", NA))))))) %>% 
    mutate(., "SampleID" = paste0(sample_name, "_X"))-> directcomp_meta_yamahara
```



## DEICODE PCA - samples compared to CN18S and CN18F

#### Load CN18S and CN18F data
```{r prepare_data_combined}
#### COI
asv_table_path_COI <- here::here("data", "COI", "ESP_CTD_COI_asv_table.csv")
tax_table_path_COI <- here::here("data", "COI", "ESP_CTD_COI_tax_table.csv")
metadata_path_COI <- here::here("data", "COI", "ESP_CTD_COI_metadata.csv")

ESP_CTD_COI_phyloseq <- merge_phyloseq(otu_table(read.csv(asv_table_path_COI, row.names = 1),taxa_are_rows = TRUE),
                                       tax_table(as.matrix(read.csv(tax_table_path_COI, row.names = 1))),
                                       sample_data(read.csv(metadata_path_COI, row.names = 1)))

ESP_CTD_COI_envt_phyloseq <- prune_samples(sample_data(ESP_CTD_COI_phyloseq)$CTD_or_ESP %in% c("CTD","ESP"),ESP_CTD_COI_phyloseq)

# Join direct comp and envt phyloseq

ESP_CTD_COI_directcomp_CN18FS_phyloseq <- merge_phyloseq(ESP_CTD_COI_envt_phyloseq, directcomp_COI_phyloseq)
```

### Prep/export data for DEICODE
```{r prep_deicode_data_directcomp_CN18SF}
# Set directory for saving DEICODE data
deicode_dir <- here::here("18S_12S_CO1_Analyses", "DEICODE", "COI_directcomp_CN18F_CN18S")

# Reformat and export OTU Table
directcomp_CN18FS_COI_asv_table <- as.data.frame(as(otu_table(ESP_CTD_COI_directcomp_CN18FS_phyloseq),"matrix"))
directcomp_CN18FS_COI_asv_table <- tibble::rownames_to_column(directcomp_CN18FS_COI_asv_table,"#OTUID")
write.table(directcomp_CN18FS_COI_asv_table, paste0(deicode_dir, "/directcomp_CN18FS_COI_asv_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_CN18FS_COI_tax_table <- as.data.frame(as(tax_table(ESP_CTD_COI_directcomp_CN18FS_phyloseq),"matrix"))
directcomp_CN18FS_COI_tax_table <- tibble::rownames_to_column(directcomp_CN18FS_COI_tax_table,"#OTUID")
write.table(directcomp_CN18FS_COI_tax_table,paste0(deicode_dir, "/directcomp_CN18FS_COI_tax_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_CN18FS_COI_metadata <- as.data.frame(as.matrix(sample_data(ESP_CTD_COI_directcomp_CN18FS_phyloseq)))
directcomp_CN18FS_COI_metadata <- tibble::rownames_to_column(directcomp_CN18FS_COI_metadata,"#SampleID")
write.table(directcomp_CN18FS_COI_metadata, paste0(deicode_dir, "/directcomp_CN18FS_COI_sample_data_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
```


### Run DEICODE in Qiime2

```{bash engine.opts = '-l'}
# Enter qiime2 environment
conda activate qiime2-2020.11

#Make biom file
biom convert -i ../DEICODE/COI_directcomp_CN18F_CN18S/directcomp_CN18FS_COI_asv_table_for_biom.txt -o ../DEICODE/COI_directcomp_CN18F_CN18S/table.from_txt_json.biom --table-type="OTU table" --to-json
#add metadata files to biom file
biom add-metadata -i ../DEICODE/COI_directcomp_CN18F_CN18S/table.from_txt_json.biom -o ../DEICODE/COI_directcomp_CN18F_CN18S/table.w_md.biom \
--observation-metadata-fp ../DEICODE/COI_directcomp_CN18F_CN18S/directcomp_CN18FS_COI_tax_table_for_biom.txt \
--sample-metadata-fp ../DEICODE/COI_directcomp_CN18F_CN18S/directcomp_CN18FS_COI_sample_data_for_biom.txt

#import into Qiime2
qiime tools import \
 --input-path ../DEICODE/COI_directcomp_CN18F_CN18S/table.w_md.biom \
 --output-path ../DEICODE/COI_directcomp_CN18F_CN18S/ESP_CTD_COI_master.biom.qza \
 --type FeatureTable[Frequency]

#run DEICODE
qiime deicode rpca \
    --i-table ../DEICODE/COI_directcomp_CN18F_CN18S/ESP_CTD_COI_master.biom.qza \
    --p-n-components 3 \
    --p-min-feature-count 10 \
    --p-min-sample-count 1000 \
    --o-biplot ../DEICODE/COI_directcomp_CN18F_CN18S/ordination.qza \
    --o-distance-matrix ../DEICODE/COI_directcomp_CN18F_CN18S/distance.qza
    
# Run PERMANOVA
# qiime diversity beta-group-significance \
#    --i-distance-matrix ../DEICODE/COI_directcomp/distance.qza \
#    --m-metadata-file ../DEICODE/COI_directcomp/directcomp_COI_sample_data_for_biom.txt \
#    --m-metadata-column sample_type.x \
#    --p-method permanova \
#    --o-visualization ../DEICODE/PERMANOVA_results/COI_ESP_CTD_PERMANOVA.qzv
```



### Plot DEICODE

```{r plot_COI_directcomp_PCA}
pco <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_directcomp_CN18F_CN18S","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

## Prepare PCA data for ggplot

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))
# Subset metadata to only those samples that had > 1000 reads and were included in DEICODE
pca_metadata <- subset(directcomp_COI_metadata, `#SampleID` %in% pca_data$SampleID)
pca_metadata %>% dplyr::rename(., "SampleID" = "#SampleID") -> pca_metadata

# Join data from Kevan Yamahara's metadata
directcomp_meta_yamahara %>% 
  right_join(., pca_data, by = "SampleID") -> pca_data

# fill in pca_data with ESP/CTD
pca_data %>% 
  mutate(Type = ifelse(is.na(Type),"CTD/ESP",Type)) %>% 
  mutate(sampling_pair = ifelse(is.na(sampling_pair),"CTD/ESP", sampling_pair)) -> pca_data

# Use shapes for ESP vs. peristaltic pump
# Use colors for each of the different pairs
pair_colors <- c("1" = "#1f78b4" , "2" = "#33a02c", "3" = "#e31a1c", "4" = "#ff7f00", "5" = "#6a3d9a", "6" = "#b15928", "CTD/ESP" = "Gray50")
type_shapes <- c("combination_depth" = 21, "bench_ESP" = 24, "CTD/ESP" = 23)


##----Run PERMANOVA-------------------------------------------------------------
# distance <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_controls","distance.qza"))
# 
# # Extract distance matrix
# distance_matrix <- distance$data
# 
# # convert DEICODE matrix to "dist" class object
# PCA_dist <- as.dist(distance_matrix)
# 
# # PERMANOVA for control vs exp
# 
# permanova_control_envt_COI <- adonis2(PCA_dist ~ sample_type.x, data = pca_metadata, permutations=999)
# permanova_control_envt_COI_value <- permanova_control_envt_COI$`Pr(>F)`[1]

##----Make plot-----------------------------------------------------------------

directcomp_COI_PCA <- ggplot(pca_data,aes(x=PC1,y=PC2,color = sampling_pair, fill = sampling_pair, shape = Type))+
  geom_point(size = 5, stroke = 2)+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = pair_colors)+
  scale_fill_manual(values = pair_colors)+
  scale_shape_manual(values = type_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1))
  # ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)+ annotate(geom = "text", x = max(pca_data$PC1), y = max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/15, label = paste0("PERMANOVA (Environment vs. Control), p = ", permanova_control_envt_COI_value), hjust = 1, size = 5)

directcomp_COI_PCA

ggsave(path = fig_dir, filename = "directcomp_CN18SCN18F_PCA.png", directcomp_COI_PCA,width = 7,height = 5)
```




```{r}
pco <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_directcomp_CN18F_CN18S","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))

## Prepare PCA data for ggplot

pca_metadata <- directcomp_CN18FS_COI_metadata
pca_metadata %>% dplyr::rename(., "SampleID" = "#SampleID") -> pca_metadata
pca_data <- pco$data$Vectors
pca_data <- right_join(pca_data,pca_metadata,on = "SampleID")
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))

# Add a categorical depth field to the metadata
pca_data  %>%
  mutate(depth = as.numeric(depth)) %>%
  mutate(depth_cat = ifelse(depth <= 25.5, "0-25",
                            ifelse(depth <= 50, "27-50",
                                   ifelse(depth <= 201 & depth >=195, "195-200",
                                          "unknown")))) -> pca_data 

pca_data %>% 
  mutate(CTD_or_ESP = ifelse(is.na(CTD_or_ESP), "dir. comp.", CTD_or_ESP)) -> pca_data


# Shape guide for depths
# depth_shapes <- c("0-25" = 21, "27-50" = 22, "195-200" = 24)
cruise_colors <- c("CN18F" = "#ff7f00", "CN18S" = "#1f78b4", "CANON17S" = "#e31a1c")
CTD_ESP_shapes <- c("CTD" = 21, "ESP" = 24, "dir. comp." = 23)
# CTD_ESP_colors = c("CTD" = "#ff7f00", "ESP" = "#1f78b4")

##----Make plot-----------------------------------------------------------------

ESP_CTD_COI_PCA <- ggplot(pca_data, aes(x=PC1,y=PC2,color = SAMPLING_cruise, fill = SAMPLING_cruise, shape = CTD_or_ESP))+
  # Make fill for geom point be white if depth is deep; if depth is shallow, then fill with season
  geom_point(size = 5, stroke = 2, fill = ifelse(pca_data$depth_cat %in% c("195-200"), "white", 
                                     ifelse(pca_data$SAMPLING_cruise == "CN18F", "#ff7f00", 
                                            ifelse(pca_data$SAMPLING_cruise == "CANON17S", "#e31a1c", "#1f78b4"))))+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = cruise_colors)+
  scale_fill_manual(values = cruise_colors)+
  scale_shape_manual(values = CTD_ESP_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1),
        legend.position = "none")+
  ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)

ESP_CTD_COI_PCA

ggsave(path = fig_dir, filename = "directcomp_COI_CN18SCN18F_PCA.png", ESP_CTD_COI_PCA,width = 7,height = 5)
```


## DEICODE PCA - samples compared to CN18S and CN18F + in situ ESP

#### Load CN18S and CN18F data
```{r prepare_directcomp_insitu_data_combined}
#### COI
asv_table_path_COI <- here::here("data", "COI", "ESP_CTD_COI_asv_table.csv")
tax_table_path_COI <- here::here("data", "COI", "ESP_CTD_COI_tax_table.csv")
metadata_path_COI <- here::here("data", "COI", "ESP_CTD_COI_metadata.csv")

ESP_CTD_COI_phyloseq <- merge_phyloseq(otu_table(read.csv(asv_table_path_COI, row.names = 1),taxa_are_rows = TRUE),
                                       tax_table(as.matrix(read.csv(tax_table_path_COI, row.names = 1))),
                                       sample_data(read.csv(metadata_path_COI, row.names = 1)))

ESP_CTD_COI_envt_phyloseq <- prune_samples(sample_data(ESP_CTD_COI_phyloseq)$CTD_or_ESP %in% c("CTD","ESP"),ESP_CTD_COI_phyloseq)

# Join direct comp + in situ ESP phyloseq and envt (CTD/ESP) phyloseq
# load asv table
directcomp_insitu_asv_table <- read.csv(here("data", "COI", "direct_comparisons", "directcomp_insitu_ESP_CTD_COI_asv_table.csv"), row.names = 1)

# load tax table
directcomp_insitu_tax_table <- read.csv(here("data", "COI", "direct_comparisons", "directcomp_insitu_ESP_CTD_COI_tax_table.csv"), row.names = 1)

# load metadata
directcomp_insitu_metadata <- read.csv(here("data", "COI", "direct_comparisons", "directcomp_insitu_ESP_CTD_COI_metadata.csv"), row.names = 1)

### Merge into phyloseq
directcomp_insitu_COI_phyloseq <- merge_phyloseq(otu_table(directcomp_insitu_asv_table,taxa_are_rows = TRUE),
                                          tax_table(as.matrix(directcomp_insitu_tax_table)),
                                          sample_data(directcomp_insitu_metadata))



ESP_CTD_COI_directcomp_insitu_CN18FS_phyloseq <- merge_phyloseq(ESP_CTD_COI_envt_phyloseq, directcomp_insitu_COI_phyloseq)
```

### Prep/export data for DEICODE
```{r prep_deicode_data_directcomp_insitu_CN18SF}
# Set directory for saving DEICODE data
deicode_dir <- here::here("18S_12S_CO1_Analyses", "DEICODE", "COI_directcomp_insitu_CN18F_CN18S")

# Reformat and export OTU Table
directcomp_insitu_CN18FS_COI_asv_table <- as.data.frame(as(otu_table(ESP_CTD_COI_directcomp_insitu_CN18FS_phyloseq),"matrix"))
directcomp_insitu_CN18FS_COI_asv_table <- tibble::rownames_to_column(directcomp_insitu_CN18FS_COI_asv_table,"#OTUID")
write.table(directcomp_insitu_CN18FS_COI_asv_table, paste0(deicode_dir, "/directcomp_insitu_CN18FS_COI_asv_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_insitu_CN18FS_COI_tax_table <- as.data.frame(as(tax_table(ESP_CTD_COI_directcomp_insitu_CN18FS_phyloseq),"matrix"))
directcomp_insitu_CN18FS_COI_tax_table <- tibble::rownames_to_column(directcomp_insitu_CN18FS_COI_tax_table,"#OTUID")
write.table(directcomp_insitu_CN18FS_COI_tax_table,paste0(deicode_dir, "/directcomp_insitu_CN18FS_COI_tax_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_insitu_CN18FS_COI_metadata <- as.data.frame(as.matrix(sample_data(ESP_CTD_COI_directcomp_insitu_CN18FS_phyloseq)))
directcomp_insitu_CN18FS_COI_metadata <- tibble::rownames_to_column(directcomp_insitu_CN18FS_COI_metadata,"#SampleID")
write.table(directcomp_insitu_CN18FS_COI_metadata, paste0(deicode_dir, "/directcomp_insitu_CN18FS_COI_sample_data_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
```


### Run DEICODE in Qiime2

```{bash engine.opts = '-l'}
# Enter qiime2 environment
conda activate qiime2-2020.11

#Make biom file
biom convert -i ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/directcomp_insitu_CN18FS_COI_asv_table_for_biom.txt -o ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/table.from_txt_json.biom --table-type="OTU table" --to-json
#add metadata files to biom file
biom add-metadata -i ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/table.from_txt_json.biom -o ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/table.w_md.biom \
--observation-metadata-fp ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/directcomp_insitu_CN18FS_COI_tax_table_for_biom.txt \
--sample-metadata-fp ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/directcomp_insitu_CN18FS_COI_sample_data_for_biom.txt

#import into Qiime2
qiime tools import \
--input-path ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/table.w_md.biom \
--output-path ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/ESP_CTD_COI_master.biom.qza \
--type FeatureTable[Frequency]

#run DEICODE
qiime deicode rpca \
--i-table ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/ESP_CTD_COI_master.biom.qza \
--p-n-components 3 \
--p-min-feature-count 10 \
--p-min-sample-count 1000 \
--o-biplot ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/ordination.qza \
--o-distance-matrix ../DEICODE/COI_directcomp_insitu_CN18F_CN18S/distance.qza

# Run PERMANOVA
# qiime diversity beta-group-significance \
#    --i-distance-matrix ../DEICODE/COI_directcomp_insitu/distance.qza \
#    --m-metadata-file ../DEICODE/COI_directcomp_insitu/directcomp_insitu_COI_sample_data_for_biom.txt \
#    --m-metadata-column sample_type.x \
#    --p-method permanova \
#    --o-visualization ../DEICODE/PERMANOVA_results/COI_ESP_CTD_PERMANOVA.qzv
```



### Plot DEICODE

```{r plot_COI_directcomp_insitu_PCA}
pco <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_directcomp_insitu_CN18F_CN18S","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

## Prepare PCA data for ggplot

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))
# Subset metadata to only those samples that had > 1000 reads and were included in DEICODE
pca_metadata <- subset(directcomp_insitu_CN18FS_COI_metadata, `#SampleID` %in% pca_data$SampleID)
pca_metadata %>% dplyr::rename(., "SampleID" = "#SampleID") -> pca_metadata

# Join data from Kevan Yamahara's metadata
directcomp_meta_yamahara %>% 
  right_join(., pca_data, by = "SampleID") -> pca_data

# fill in pca_data with ESP/CTD
pca_data %>% 
  mutate(Type = ifelse(is.na(Type),"CTD/ESP",Type)) %>% 
  mutate(sampling_pair = ifelse(is.na(sampling_pair),"CTD/ESP", sampling_pair)) -> pca_data

# Use shapes for ESP vs. peristaltic pump
# Use colors for each of the different pairs
pair_colors <- c("1" = "#1f78b4" , "2" = "#33a02c", "3" = "#e31a1c", "4" = "#ff7f00", "5" = "#6a3d9a", "6" = "#b15928", "CTD/ESP" = "Gray50")
type_shapes <- c("combination_depth" = 21, "bench_ESP" = 24, "CTD/ESP" = 23, "3G ESP" = 22)


##----Run PERMANOVA-------------------------------------------------------------
# distance <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_controls","distance.qza"))
# 
# # Extract distance matrix
# distance_matrix <- distance$data
# 
# # convert DEICODE matrix to "dist" class object
# PCA_dist <- as.dist(distance_matrix)
# 
# # PERMANOVA for control vs exp
# 
# permanova_control_envt_COI <- adonis2(PCA_dist ~ sample_type.x, data = pca_metadata, permutations=999)
# permanova_control_envt_COI_value <- permanova_control_envt_COI$`Pr(>F)`[1]

##----Make plot-----------------------------------------------------------------

directcomp_insitu_COI_PCA <- ggplot(pca_data,aes(x=PC1,y=PC2,color = sampling_pair, fill = sampling_pair, shape = Type))+
  geom_point(size = 5, stroke = 2)+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = pair_colors)+
  scale_fill_manual(values = pair_colors)+
  scale_shape_manual(values = type_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1))
# ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)+ annotate(geom = "text", x = max(pca_data$PC1), y = max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/15, label = paste0("PERMANOVA (Environment vs. Control), p = ", permanova_control_envt_COI_value), hjust = 1, size = 5)

directcomp_insitu_COI_PCA

ggsave(path = fig_dir, filename = "directcomp_insitu_CN18SCN18F_PCA.png", directcomp_insitu_COI_PCA,width = 7,height = 5)
```




```{r}
pco <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_directcomp_insitu_CN18F_CN18S","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))

## Prepare PCA data for ggplot

pca_metadata <- directcomp_insitu_CN18FS_COI_metadata
pca_metadata %>% dplyr::rename(., "SampleID" = "#SampleID") -> pca_metadata
pca_data <- pco$data$Vectors
pca_data <- right_join(pca_data,pca_metadata,on = "SampleID")
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))

# Add a categorical depth field to the metadata
pca_data  %>%
  mutate(depth = as.numeric(depth)) %>%
  mutate(depth_cat = ifelse(depth <= 25.5, "0-25",
                            ifelse(depth <= 50, "27-50",
                                   ifelse(depth <= 201 & depth >=195, "195-200",
                                          "unknown")))) -> pca_data 

pca_data %>% 
  mutate(CTD_or_ESP = ifelse(samp_collection_device == "3GESP" & SAMPLING_cruise == "CANON17S", "in situ ESP CN17S",
         ifelse(is.na(CTD_or_ESP), "dir. comp.", CTD_or_ESP))) -> pca_data


# Shape guide for depths
# depth_shapes <- c("0-25" = 21, "27-50" = 22, "195-200" = 24)
cruise_colors <- c("CN18F" = "#ff7f00", "CN18S" = "#1f78b4", "CANON17S" = "#e31a1c")
CTD_ESP_shapes <- c("CTD" = 21, "ESP" = 24, "dir. comp." = 23, "in situ ESP CN17S" = 22)
# CTD_ESP_colors = c("CTD" = "#ff7f00", "ESP" = "#1f78b4")

##----Make plot-----------------------------------------------------------------

ESP_CTD_COI_PCA <- ggplot(pca_data, aes(x=PC1,y=PC2,color = SAMPLING_cruise, fill = SAMPLING_cruise, shape = CTD_or_ESP))+
  # Make fill for geom point be white if depth is deep; if depth is shallow, then fill with season
  geom_point(size = 5, stroke = 2, fill = ifelse(pca_data$depth_cat %in% c("195-200"), "white", 
                                                 ifelse(pca_data$SAMPLING_cruise == "CN18F", "#ff7f00", 
                                                        ifelse(pca_data$SAMPLING_cruise == "CANON17S", "#e31a1c", "#1f78b4"))))+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = cruise_colors)+
  scale_fill_manual(values = cruise_colors)+
  scale_shape_manual(values = CTD_ESP_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1),
        legend.position = "none")+
  ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)

ESP_CTD_COI_PCA

ggsave(path = fig_dir, filename = "directcomp_insitu_COI_CN18SCN18F_PCA.png", ESP_CTD_COI_PCA,width = 7,height = 5)
```





## DEICODE PCA - direct comparison only

### Prep data for DEICODE
```{r prep_deicode_data}
# Keep only the samples that are definitely labeled correctly
directcomp_COI_phyloseq <- subset_samples(directcomp_COI_phyloseq, sample_names(directcomp_COI_phyloseq) %in% c("CANON_B10_22_X", "CANON_B13_22_X",   "CANON_B17_22_X",   "CANON_B1_22_X",  "CANON_CBI1_22_X", "CANON_CBI4_22_X",  "CANON_CBI5_22_X",  "CANON_CBI8_22_X"))
                                        
# Set directory for saving DEICODE data
deicode_dir <- here::here("18S_12S_CO1_Analyses", "DEICODE", "COI_directcomp")

# Reformat and export OTU Table
directcomp_COI_asv_table <- as.data.frame(as(otu_table(directcomp_COI_phyloseq),"matrix"))
directcomp_COI_asv_table <- tibble::rownames_to_column(directcomp_COI_asv_table,"#OTUID")
write.table(directcomp_COI_asv_table, paste0(deicode_dir, "/directcomp_COI_asv_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_COI_tax_table <- as.data.frame(as(tax_table(directcomp_COI_phyloseq),"matrix"))
directcomp_COI_tax_table <- tibble::rownames_to_column(directcomp_COI_tax_table,"#OTUID")
write.table(directcomp_COI_tax_table,paste0(deicode_dir, "/directcomp_COI_tax_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_COI_metadata <- as.data.frame(as.matrix(sample_data(directcomp_COI_phyloseq)))
directcomp_COI_metadata <- tibble::rownames_to_column(directcomp_COI_metadata,"#SampleID")
write.table(directcomp_COI_metadata, paste0(deicode_dir, "/directcomp_COI_sample_data_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
```

### Run DEICODE in Qiime2

```{bash engine.opts = '-l'}
# Enter qiime2 environment
conda activate qiime2-2020.11

#Make biom file
biom convert -i ../DEICODE/COI_directcomp/directcomp_COI_asv_table_for_biom.txt -o ../DEICODE/COI_directcomp/table.from_txt_json.biom --table-type="OTU table" --to-json
#add metadata files to biom file
biom add-metadata -i ../DEICODE/COI_directcomp/table.from_txt_json.biom -o ../DEICODE/COI_directcomp/table.w_md.biom \
--observation-metadata-fp ../DEICODE/COI_directcomp/directcomp_COI_tax_table_for_biom.txt \
--sample-metadata-fp ../DEICODE/COI_directcomp/directcomp_COI_sample_data_for_biom.txt

#import into Qiime2
qiime tools import \
 --input-path ../DEICODE/COI_directcomp/table.w_md.biom \
 --output-path ../DEICODE/COI_directcomp/ESP_CTD_COI_master.biom.qza \
 --type FeatureTable[Frequency]

#run DEICODE
qiime deicode rpca \
    --i-table ../DEICODE/COI_directcomp/ESP_CTD_COI_master.biom.qza \
    --p-n-components 3 \
    --p-min-feature-count 10 \
    --p-min-sample-count 1000 \
    --o-biplot ../DEICODE/COI_directcomp/ordination.qza \
    --o-distance-matrix ../DEICODE/COI_directcomp/distance.qza
    
# Run PERMANOVA
# qiime diversity beta-group-significance \
#    --i-distance-matrix ../DEICODE/COI_directcomp/distance.qza \
#    --m-metadata-file ../DEICODE/COI_directcomp/directcomp_COI_sample_data_for_biom.txt \
#    --m-metadata-column sample_type.x \
#    --p-method permanova \
#    --o-visualization ../DEICODE/PERMANOVA_results/COI_ESP_CTD_PERMANOVA.qzv
```

### Plot DEICODE

```{r plot_COI_directcomp_PCA}
pco <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_directcomp","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

## Prepare PCA data for ggplot

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))
# Subset metadata to only those samples that had > 1000 reads and were included in DEICODE
pca_metadata <- subset(directcomp_COI_metadata, `#SampleID` %in% pca_data$SampleID)
pca_metadata %>% dplyr::rename(., "SampleID" = "#SampleID") -> pca_metadata

# Join data from Kevan Yamahara's metadata
directcomp_meta_yamahara %>% 
  right_join(., pca_data, by = "SampleID") -> pca_data

# Use shapes for ESP vs. peristaltic pump
# Use colors for each of the different pairs
pair_colors <- c("1" = "#1f78b4" , "2" = "#33a02c", "3" = "#e31a1c", "4" = "#ff7f00", "5" = "#6a3d9a", "6" = "#b15928")
# type_shapes <- c("combination_depth" = 21, "bench_ESP" = 24)
type_shapes <- c("combination_depth" = 19, "bench_ESP" = 17)


##----Run PERMANOVA-------------------------------------------------------------
# distance <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_controls","distance.qza"))
# 
# # Extract distance matrix
# distance_matrix <- distance$data
# 
# # convert DEICODE matrix to "dist" class object
# PCA_dist <- as.dist(distance_matrix)
# 
# # PERMANOVA for control vs exp
# 
# permanova_control_envt_COI <- adonis2(PCA_dist ~ sample_type.x, data = pca_metadata, permutations=999)
# permanova_control_envt_COI_value <- permanova_control_envt_COI$`Pr(>F)`[1]

##----Make plot-----------------------------------------------------------------


# Move one sample (pair 2, bench ESP/CANON_CBI4_22)
pca_data %>%
  mutate(., PC1 = ifelse(sample_name == "CANON_CBI4_22", PC1 + 0.02, PC1)) %>% 
  mutate(., PC2 = ifelse(sample_name == "CANON_CBI4_22", PC2 + 0.02, PC2)) -> pca_data

directcomp_COI_PCA <- ggplot(pca_data,aes(x=PC1,y=PC2,color = sampling_pair,  shape = Type))+
  geom_point(size = 5, stroke = 2)+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  labs(color = "Sampling Pair", shape = "Filtration Method") +
  scale_color_manual(values = pair_colors)+
  # scale_fill_manual(values = pair_colors)+
  scale_shape_manual(values = type_shapes, labels = c("Benchtop ESP", "Peristaltic Pump"))+
  # guides(color=guide_legend(override.aes=list(fill=NA)), shape=guide_legend(override.aes=list(fill=NA))) +
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 15),
        legend.position = "top",
        legend.direction = "horizontal",
        legend.box = "vertical",
        legend.key = element_rect(fill = NA))
  # ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)+ annotate(geom = "text", x = max(pca_data$PC1), y = max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/15, label = paste0("PERMANOVA (Environment vs. Control), p = ", permanova_control_envt_COI_value), hjust = 1, size = 5)

directcomp_COI_PCA

ggsave(path = fig_dir, filename = "directcomp_COI_PCA_2.png", directcomp_COI_PCA,width = 7,height = 5)
```

## Compare taxa

### Plot direct comparison taxa - top 20 ASVs

#### Make a function for plotting
```{r plot_top_20ASVs_function}
plot_top20ASVs_directcomp <- function(sample_names, plot_name){
  # Subset the samples specified in the function
  phyloseq <- prune_samples(sample_names(directcomp_COI_phyloseq) %in% sample_names, directcomp_COI_phyloseq)
  
  directcomp_COI_asv_table <- as.data.frame(otu_table(phyloseq))
directcomp_COI_asv_table <- tibble::rownames_to_column(directcomp_COI_asv_table,"ASV")

directcomp_COI_tax_table <- as.data.frame(as(tax_table(phyloseq),"matrix"))
directcomp_COI_tax_table <- tibble::rownames_to_column(directcomp_COI_tax_table,"ASV")
directcomp_COI_tax_table %>% mutate_all(as.character) -> directcomp_COI_tax_table

# Rename ASVs with their taxonomy
directcomp_COI_asv_table <- left_join(directcomp_COI_asv_table, directcomp_COI_tax_table, by = "ASV")

directcomp_COI_asv_table %>% mutate(.,ASV_name = ifelse(!(Species %in% c("s_","unassigned")),paste0(ASV,"_",Species),
                                                    ifelse(!(Genus %in% c("g_","unassigned")),paste0(ASV,"_",Genus),
                                                           ifelse(!(Family == "unassigned"),paste0(ASV,"_",Family),
                                                                  ifelse(!(Order == "unassigned"),paste0(ASV,"_",Order),
                                                                         ifelse(!(Class == "unassigned"),paste0(ASV,"_",Class),
                                                                                ifelse(!(Phylum == "unassigned"),paste0(ASV,"_",Phylum),
                                                                                       paste0(ASV,"_","unassigned")))))))) -> directcomp_COI_asv_table
# Convert to long form
gathercols <- colnames(as.data.frame(otu_table(phyloseq)))

directcomp_COI_asv_table_gather <- gather(directcomp_COI_asv_table, key = "sample", value = "reads", gathercols, factor_key=TRUE)


# Figure out the top 20 ASVs total in the dataset to plot
directcomp_COI_asv_table_asv_total_reads <- as.data.frame(sort(-taxa_sums(phyloseq)))
top_19_CN18S_blanks_asvs <- rownames(directcomp_COI_asv_table_asv_total_reads)[1:19]

directcomp_COI_asv_table_gather %>% mutate(.,ASV_name_updated = ifelse(ASV %in% top_19_CN18S_blanks_asvs,ASV_name,"other")) -> directcomp_COI_asv_table_gather_updated

# Add metadata
directcomp_meta_yamahara %>% 
  dplyr::rename(sample = SampleID) %>% 
  right_join(., directcomp_COI_asv_table_gather_updated, by = "sample") %>% 
  mutate(Type = ifelse(Type == "combination_depth", "peristaltic_pump", "benchtop_ESP")) %>% 
  mutate(pair = paste0(sampling_pair, "_", Type)) -> directcomp_COI_asv_table_gather_updated

directcomp_COI_gg <- ggplot(directcomp_COI_asv_table_gather_updated,aes(x = pair, y = reads, fill = ASV_name_updated))+
  geom_bar(stat = "identity")+
  ggtitle(plot_name)+
  scale_fill_tableau("Tableau 20")+
  # ylim(0,20000)+
  theme(legend.text = element_text(size = 7),
        axis.text.x = element_text(angle = 90),
        axis.title.x = element_blank())+
  guides(fill = guide_legend(title = "ASV"))

directcomp_COI_gg
return(directcomp_COI_gg)
ggsave(paste0(fig_dir, "/", plot_name, ".png"), directcomp_COI_gg, height = 6.5, width = 8)
}

```


#### All samples together
```{r}
directcomp_COI_asv_table <- as.data.frame(otu_table(directcomp_COI_phyloseq))
directcomp_COI_asv_table <- tibble::rownames_to_column(directcomp_COI_asv_table,"ASV")

directcomp_COI_tax_table <- as.data.frame(as(tax_table(directcomp_COI_phyloseq),"matrix"))
directcomp_COI_tax_table <- tibble::rownames_to_column(directcomp_COI_tax_table,"ASV")
directcomp_COI_tax_table %>% mutate_all(as.character) -> directcomp_COI_tax_table

# Rename ASVs with their taxonomy
directcomp_COI_asv_table <- left_join(directcomp_COI_asv_table, directcomp_COI_tax_table, by = "ASV")

directcomp_COI_asv_table %>% mutate(.,ASV_name = ifelse(!(Species %in% c("s_","unassigned")),paste0(ASV,"_",Species),
                                                    ifelse(!(Genus %in% c("g_","unassigned")),paste0(ASV,"_",Genus),
                                                           ifelse(!(Family == "unassigned"),paste0(ASV,"_",Family),
                                                                  ifelse(!(Order == "unassigned"),paste0(ASV,"_",Order),
                                                                         ifelse(!(Class == "unassigned"),paste0(ASV,"_",Class),
                                                                                ifelse(!(Phylum == "unassigned"),paste0(ASV,"_",Phylum),
                                                                                       paste0(ASV,"_","unassigned")))))))) -> directcomp_COI_asv_table
# Convert to long form
gathercols <- colnames(as.data.frame(otu_table(directcomp_COI_phyloseq)))

directcomp_COI_asv_table_gather <- gather(directcomp_COI_asv_table, key = "sample", value = "reads", gathercols, factor_key=TRUE)


# Figure out the top 20 ASVs total in the dataset to plot
directcomp_COI_asv_table_asv_total_reads <- as.data.frame(sort(-taxa_sums(directcomp_COI_phyloseq)))
top_19_CN18S_blanks_asvs <- rownames(directcomp_COI_asv_table_asv_total_reads)[1:19]

directcomp_COI_asv_table_gather %>% mutate(.,ASV_name_updated = ifelse(ASV %in% top_19_CN18S_blanks_asvs,ASV_name,"other")) -> directcomp_COI_asv_table_gather_updated

# Add metadata
directcomp_meta_yamahara %>% 
  dplyr::rename(sample = SampleID) %>% 
  right_join(., directcomp_COI_asv_table_gather_updated, by = "sample") %>% 
  mutate(Type = ifelse(Type == "combination_depth", "peristaltic_pump", "benchtop_ESP")) %>% 
  mutate(pair = paste0(sampling_pair, "_", Type)) -> directcomp_COI_asv_table_gather_updated

directcomp_COI_gg <- ggplot(directcomp_COI_asv_table_gather_updated,aes(x = pair, y = reads, fill = ASV_name_updated))+
  geom_bar(stat = "identity")+
  ggtitle("Direct comparisons")+
  scale_fill_tableau("Tableau 20")+
  # ylim(0,20000)+
  theme(legend.text = element_text(size = 7),
        axis.text.x = element_text(angle = 90),
        axis.title.x = element_blank())+
  guides(fill = guide_legend(title = "ASV"))

directcomp_COI_gg

ggsave(paste0(fig_dir, "/directcomp_COI_top20ASVs.png"), directcomp_COI_gg, height = 6.5, width = 8)
```

#### Plot individual sample pairs
```{r plot_sample_pairs}
plot_top20ASVs_directcomp(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "1")$SampleID, plot_name = "directcomp_pair1")

plot_top20ASVs_directcomp(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "2")$SampleID, plot_name = "directcomp_pair2")

plot_top20ASVs_directcomp(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "3")$SampleID, plot_name = "directcomp_pair3")

plot_top20ASVs_directcomp(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "4")$SampleID, plot_name = "directcomp_pair4")

plot_top20ASVs_directcomp(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "5")$SampleID, plot_name = "directcomp_pair5")

plot_top20ASVs_directcomp(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "6")$SampleID, plot_name = "directcomp_pair6")
```





## Plot direct comparison taxa: inverts that might have gotten stuck on filters

Compare arthropods?
```{r arthropod_comparison}
# Use long form data from earlier
directcomp_COI_asv_table_gather %>% 
  mutate(sample = as.character(sample)) %>% 
  mutate(phylum2 = ifelse(Phylum == "Arthropoda", "Arthropoda", "other")) -> directcomp_COI_asv_table_gather2

# Add metadata
directcomp_meta_yamahara %>% 
  dplyr::rename(sample = SampleID) %>% 
  right_join(., directcomp_COI_asv_table_gather2, by = "sample") %>% 
  mutate(Type = ifelse(Type == "combination_depth", "peristaltic_pump", "benchtop_ESP")) %>% 
  mutate(pair = paste0(sampling_pair, "_", Type)) -> directcomp_COI_asv_table_gather2

```

##### Make a plotting function for reads
```{r}
fig_dir <- here("figures", "direct_comparisons", "arthropods")
compare_arthropods_plot <- function(sample_names, plot_name){
  data <- subset(directcomp_COI_asv_table_gather2, sample %in% sample_names)
  
  directcomp_COI_gg <- ggplot(data,aes(x = pair, y = reads, fill = phylum2))+
  geom_bar(stat = "identity")+
  ggtitle(plot_name)+
  scale_fill_tableau("Tableau 10")+
  # ylim(0,20000)+
  theme(legend.text = element_text(size = 7),
        axis.text.x = element_text(angle = 90),
        axis.title.x = element_blank())+
  guides(fill = guide_legend(title = "Phylum"))

# directcomp_COI_gg
# return(directcomp_COI_gg)
ggsave(paste0(fig_dir, "/", plot_name, ".png"), directcomp_COI_gg, height = 6.5, width = 8)
}

# Plot them all!!
compare_arthropods_plot(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "1")$SampleID, plot_name = "directcomp1_arthropods")

compare_arthropods_plot(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "2")$SampleID, plot_name = "directcomp2_arthropods")

compare_arthropods_plot(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "3")$SampleID, plot_name = "directcomp3_arthropods")

compare_arthropods_plot(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "4")$SampleID, plot_name = "directcomp4_arthropods")

compare_arthropods_plot(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "5")$SampleID, plot_name = "directcomp5_arthropods")

compare_arthropods_plot(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "6")$SampleID, plot_name = "directcomp6_arthropods")

```

### Number of ASVs
```{r}
directcomp_COI_asv_table_gather2 %>% 
  subset(reads > 0) %>% 
  # group_by(sample, phylum2) %>%  
  # summarise(nASVs = n(ASV))
  count(sample, phylum2) %>% 
  dplyr::rename(SampleID = sample) %>% 
  left_join(., directcomp_meta_yamahara, by = "SampleID") %>% 
  dplyr::rename(sample = SampleID) %>% 
  mutate(Type = ifelse(Type == "combination_depth", "peristaltic_pump", "benchtop_ESP")) %>% 
  mutate(pair = paste0(sampling_pair, "_", Type)) -> directcomp_COI_asv_count
```


##### Make a plotting function for number of ASVs
```{r}
fig_dir <- here("figures", "direct_comparisons", "arthropods", "nASVs")
compare_arthropods_plot_nASVs <- function(sample_names, plot_name){
  data <- subset(directcomp_COI_asv_count, sample %in% sample_names)
  
  directcomp_COI_gg <- ggplot(data,aes(x = pair, y = n, fill = phylum2))+
  geom_bar(stat = "identity")+
  ggtitle(plot_name)+
  scale_fill_tableau("Tableau 10")+
    ylab("Number of ASVs")+ 
  # ylim(0,20000)+
  theme(legend.text = element_text(size = 7),
        axis.text.x = element_text(angle = 90),
        axis.title.x = element_blank())+
  guides(fill = guide_legend(title = "Phylum"))

# directcomp_COI_gg
# return(directcomp_COI_gg)
ggsave(paste0(fig_dir, "/", plot_name, ".png"), directcomp_COI_gg, height = 6.5, width = 8)
}

# Plot them all!!
compare_arthropods_plot_nASVs(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "1")$SampleID, plot_name = "directcomp1_arthropods_nASVs")

compare_arthropods_plot_nASVs(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "2")$SampleID, plot_name = "directcomp2_arthropods_nASVs")

compare_arthropods_plot_nASVs(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "3")$SampleID, plot_name = "directcomp3_arthropods_nASVs")

compare_arthropods_plot_nASVs(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "4")$SampleID, plot_name = "directcomp4_arthropods_nASVs")

compare_arthropods_plot_nASVs(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "5")$SampleID, plot_name = "directcomp5_arthropods_nASVs")

compare_arthropods_plot_nASVs(sample_names = subset(directcomp_meta_yamahara, sampling_pair == "6")$SampleID, plot_name = "directcomp6_arthropods_nASVs")

```


## Make plot for figure

### Make top phyla plot
```{r prep_data}
# Extract ASVs
as.data.frame(otu_table(directcomp_COI_phyloseq)) %>% 
  rownames_to_column("ASV") -> directcomp_COI_ASV_table

# Remove "_X" from the column names
colnames(directcomp_COI_ASV_table) <- gsub("_X", "", colnames(directcomp_COI_ASV_table))

# Extract taxonomy table
directcomp_COI_tax_table <- as.data.frame(as(tax_table(directcomp_COI_phyloseq),"matrix"))
directcomp_COI_tax_table <- tibble::rownames_to_column(directcomp_COI_tax_table,"ASV")

# reformat ASV table as long form
directcomp_COI_ASV_table %>% 
  pivot_longer(cols = colnames(directcomp_COI_ASV_table)[2:ncol(directcomp_COI_ASV_table)], names_to = "sample_name") -> directcomp_COI_asv_table_long

directcomp_COI_asv_table_long %>% 
  left_join(directcomp_COI_tax_table, by = "ASV") -> DC_COI_table

# Add shortened sample names
directcomp_meta_yamahara %>% 
  mutate(name = paste0(sampling_pair, ": ", ifelse(Type == "combination_depth", "peristaltic \n pump", 
                                                   ifelse(Type == "bench_ESP", "benchtop \n ESP",Type)))) -> directcomp_meta_yamahara

DC_COI_table %>% 
  left_join(., dplyr::select(directcomp_meta_yamahara, sample_name, name, Type, sampling_pair), by = "sample_name") -> DC_COI_table

# Keep only top four pairs (due to likely sample mislabeling with pairs 5 and 6)
DC_COI_table <- subset(DC_COI_table, sampling_pair %in% c("1", "2", "3", "4"))
```

```{r determine_top_phyla}
DC_COI_table %>% 
  # Change so that "unassigned", "no_hit", and "unknown" will merge together
  mutate(., Phylum = ifelse(Phylum %in% c("unassigned", "no_hit", "unknown"), "Unassigned", Phylum)) %>% 
  group_by(Phylum) %>% 
  summarise(total = sum(value)) -> DC_COI_phyla_reads

DC_COI_phyla_reads %>% 
  arrange(., desc(total)) -> DC_COI_phyla_reads

# Get the top ten phyla; we need top ten, then "unknown" and "other" to get to 12
# Get top 11
top11_phyla <- DC_COI_phyla_reads$Phylum[1:11]

# Mutate data for plotting to get new phyla (top 10 + unassigned + other)
DC_COI_table %>% 
  mutate(., Phylum = ifelse(Phylum %in% c("unassigned", "no_hit", "unknown"), "Unassigned", Phylum)) %>% 
  mutate(., phylum2 = ifelse(Phylum %in% top11_phyla, Phylum, "Other")) -> DC_COI_table

# Calculate proportion of total
DC_COI_table %>% 
  group_by(name) %>% 
  mutate(rel_reads = value/sum(value)*100) -> DC_COI_table

# Group by phylum and collapse table
DC_COI_table %>% 
  group_by(name, phylum2) %>% 
  summarise(rel_reads = sum(rel_reads)) -> DC_COI_table_forplot

# Reorder phyla so that unassigned and other are at the bottom
DC_COI_table_forplot$phylum2 <- factor(DC_COI_table_forplot$phylum2, levels = c("Arthropoda", "Bacillariophyta", "Cercozoa", "Chlorophyta",
                                                                                "Cnidaria", "Haptophyta", "Ochrophyta", "Oomycota",
                                                                                "Picozoa", "Rotifera", "Other", "Unassigned"))

# Reorder samples
DC_COI_table_forplot$name <- factor(DC_COI_table_forplot$name, levels = c("4: benchtop \n ESP", "4: peristaltic \n pump",
                                                                          "3: benchtop \n ESP", "3: peristaltic \n pump",
                                                                          "2: benchtop \n ESP", "2: peristaltic \n pump",
                                                                          "1: benchtop \n ESP", "1: peristaltic \n pump"))

# Make color scale
phylum_colors <- c("#a6cee3",
"#1f78b4",
"#b2df8a",
"#33a02c",
'#fb9a99',
"#e31a1c",
"#fdbf6f",
"#ff7f00",
"#cab2d6",
"#6a3d9a",
"#b15928",
"gray50")
# "#ffff99",
# "#b15928")
```

```{r plot_top_phyla}
DC_COI_top_phyla_plot <- ggplot(DC_COI_table_forplot, aes(x = name, y = rel_reads, fill = phylum2)) +
  geom_bar(stat = "identity", color = "black", size = 0.3) +
  scale_fill_manual(values = phylum_colors)+
  labs(x = "Sample", y = "Percentage of total reads") +
  scale_y_reverse(expand = c(0,0), labels = c("100", "75", "50", "25", "0")) +
  coord_flip()+
  scale_x_discrete(expand = c(0,0)) + 
  guides(fill = guide_legend(title = "Phylum", nrow = 4, byrow = TRUE, title.position = "top")) +
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title.x = element_text(size = 14),
        # Remove the axis labels (sample names)so that there is only one set of labels in the combined 16S/COI figure
        axis.text.y = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks.length.y=unit(0, "cm"),
        axis.ticks.length.x = unit(0.1, "cm"),
        # axis.ticks=element_blank(),
        plot.margin = margin(0.5, 0.25, 0.25, 0.25, "cm"),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 15),
        legend.position = "top",
        legend.title.align = 0.5)
        # axis.line.x.bottom = element_line(color = "black", size = 1),
        # axis.line.y.left = element_line(color = "black", size = 1))

DC_COI_top_phyla_plot

# ggsave(paste0(fig_dir, "/COI_directcomp_phyla_plot.png"), DC_COI_top_phyla_plot, height = 8, width = 6)



# Second version of plot with shapes as y axis labels

# # Add sampling pair and filtration method to table
# DC_shapes_y_axis = data.frame(name = c("1: peristaltic pump", "1: benchtop ESP", 
#                                        "2: peristaltic pump", "2: benchtop ESP", 
#                                        "3: peristaltic pump", "3: benchtop ESP", 
#                                        "4: peristaltic pump", "4: benchtop ESP"),
#                               filtration_method = c("benchtop ESP", "peristaltic pump",
#                                        "benchtop ESP", "peristaltic pump",
#                                        "benchtop ESP", "peristaltic pump",
#                                        "benchtop ESP", "peristaltic pump"),
#                               sampling_pair = as.character(c(1,1,2,2,3,3,4,4)),
#                               # pos.x = c(-1,-1,-1,-1,-1,-1,-1,-1),
#                               pos.x = rep(105,8),
#                               pos.y = c(8,7,6,5,4,3,2,1))
# 
# filt_method_shapes <- c("benchtop ESP" = 17,	"peristaltic pump" = 19)
# 
# DC_COI_top_phyla_plot <- ggplot(DC_COI_table_forplot, aes(x = name, y = rel_reads, fill = phylum2)) +
#   geom_bar(stat = "identity", color = "black", size = 0.3) +
#   scale_fill_manual(values = phylum_colors)+
#   labs(x = "Sample", y = "Percentage of total reads") +
#   scale_y_reverse(expand = c(0,0), labels = c("100", "75", "50", "25", "0"), limits = c(110, -5)) +
#   coord_flip()+
#   scale_x_discrete(expand = c(0,0)) + 
#   guides(fill = guide_legend(title = "Phylum", nrow = 3, byrow = TRUE, title.position = "top")) +
#   theme(panel.background = element_rect(fill = "white",size = 1),
#         panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
#         axis.title = element_text(size = 14),
#         # axis.text = element_blank(),
#         axis.ticks.length.y=unit(0, "cm"),
#         axis.ticks.length.x = unit(0.1, "cm"),
#         axis.text.y = element_blank(),
#         axis.text.x = element_text(size = 12),
#         # axis.ticks=element_blank(),
#         # plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
#         legend.text = element_text(size = 12),
#         legend.title = element_text(size = 15),
#         legend.position = "top",
#         legend.title.align = 0.5)+
#   # Add the points 
#   geom_point(data = DC_shapes_y_axis, inherit.aes = FALSE, aes(x = pos.y, y = pos.x, shape = filtration_method, color = sampling_pair), size = 6) +
#     scale_color_manual(values = pair_colors)+
#   scale_shape_manual(values = filt_method_shapes)+
#   guides(shape = FALSE, color = FALSE)
#         # axis.line.x.bottom = element_line(color = "black", size = 1),
#         # axis.line.y.left = element_line(color = "black", size = 1))
# 
# DC_COI_top_phyla_plot

```

### Join top phyla plot with DEICODE PCA using ggarrange
```{r make_directcomp_fig}
DC_COI_fig <- ggarrange(directcomp_COI_PCA, DC_COI_top_phyla_plot, heights = c(7,7), widths = c(5,5))

ggsave(paste0(fig_dir, "/final_figures/COI_directcomp_fig_v2.png"), DC_COI_fig, height = 8, width = 12)
```




# 16S


### load data
```{r load_data}
# load asv table
read.table(here("data", "16S", "2017_directcomparison_tablewithtaxonomy.tsv"), sep = "\t") %>% 
  row_to_names(row_number = 1) %>% 
  dplyr::rename(ASV_ID = `ASV ID`) -> asv_tax_table_16S


# Split off taxonomy
asv_tax_table_16S %>% 
  dplyr::select(-Taxon) %>% 
  # mutate_at(vars(!(contains("ASV_ID"))), funs(as.numeric())) -> asv_table_16S
  mutate_if(!(names(.) == "ASV_ID"), funs(as.numeric(.))) -> asv_table_16S
rownames(asv_table_16S) <- NULL

asv_tax_table_16S %>% 
  dplyr::select(ASV_ID, Taxon) %>% 
  separate(Taxon, c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep = "; ") %>% 
  mutate_all(funs(sub(".*__", "", .))) -> tax_table_16S
rownames(tax_table_16S) <- NULL
  

### Load sample metadata (Kevan Yamahara's corrected data)
read_excel(here("data", "ESP_CTD_matched_samples_metadata_YAMAHARACORRECTED.xlsx"), sheet = "CN17S") %>% 
  # rename sample names to match 16S
  mutate(sample_name = sub("_[^_]+$", "", sample_name)) %>% 
  mutate(sample_name = sub("CANON", "CN17", sample_name)) %>% 
  subset(Type %in% c("combination_depth", "bench_ESP")) -> yamahara_metadata_16S


### Merge into phyloseq
directcomp_16S_phyloseq <- merge_phyloseq(otu_table(column_to_rownames(asv_table_16S, "ASV_ID"),taxa_are_rows = TRUE),
                                       tax_table(as.matrix(column_to_rownames(tax_table_16S, "ASV_ID"))),
                                       sample_data(column_to_rownames(yamahara_metadata_16S, "sample_name")))


# Add another column to show which samples are paired
yamahara_metadata_16S %>% 
  mutate(., sampling_pair = ifelse(sample_name %in% c("CN17_B1", "CN17_CBI1"), "1",
                                   ifelse(sample_name %in% c("CN17_B10", "CN17_CBI4"), "2",
                                          ifelse(sample_name %in% c("CN17_B13", "CN17_CBI5"), "3",
                                                 ifelse(sample_name %in% c("CN17_B17", "CN17_CBI8"), "4",
                                                        ifelse(sample_name %in% c("CN17_B20", "CN17_CBI9"), "5",
                                                               ifelse(sample_name %in% c("CN17_B23", "CN17_CBI10"), "6", NA))))))) -> yamahara_metadata_16S
```



## DEICODE PCA - direct comparison only

### Prep data for DEICODE
```{r prep_deicode_data}
# Set directory for saving DEICODE data
deicode_dir <- here::here("16S_Analyses", "DEICODE", "16S_directcomp")

# Reformat and export OTU Table
directcomp_16S_asv_table <- as.data.frame(as(otu_table(directcomp_16S_phyloseq),"matrix"))
directcomp_16S_asv_table <- tibble::rownames_to_column(directcomp_16S_asv_table,"#OTUID")
write.table(directcomp_16S_asv_table, paste0(deicode_dir, "/directcomp_16S_asv_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_16S_tax_table <- as.data.frame(as(tax_table(directcomp_16S_phyloseq),"matrix"))
directcomp_16S_tax_table <- tibble::rownames_to_column(directcomp_16S_tax_table,"#OTUID")
write.table(directcomp_16S_tax_table,paste0(deicode_dir, "/directcomp_16S_tax_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_16S_metadata <- as.data.frame(as.matrix(sample_data(directcomp_16S_phyloseq)))
directcomp_16S_metadata <- tibble::rownames_to_column(directcomp_16S_metadata,"#SampleID")
write.table(directcomp_16S_metadata, paste0(deicode_dir, "/directcomp_16S_sample_data_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
```

### Run DEICODE in Qiime2

```{bash engine.opts = '-l'}
# Enter qiime2 environment
conda activate qiime2-2020.11

#Make biom file
biom convert -i ../../16S_Analyses/DEICODE/16S_directcomp/directcomp_16S_asv_table_for_biom.txt -o ../../16S_Analyses/DEICODE/16S_directcomp/table.from_txt_json.biom --table-type="OTU table" --to-json
#add metadata files to biom file
biom add-metadata -i ../../16S_Analyses/DEICODE/16S_directcomp/table.from_txt_json.biom -o ../../16S_Analyses/DEICODE/16S_directcomp/table.w_md.biom \
--observation-metadata-fp ../../16S_Analyses/DEICODE/16S_directcomp/directcomp_16S_tax_table_for_biom.txt \
--sample-metadata-fp ../../16S_Analyses/DEICODE/16S_directcomp/directcomp_16S_sample_data_for_biom.txt

#import into Qiime2
qiime tools import \
 --input-path ../../16S_Analyses/DEICODE/16S_directcomp/table.w_md.biom \
 --output-path ../../16S_Analyses/DEICODE/16S_directcomp/ESP_CTD_16S_master.biom.qza \
 --type FeatureTable[Frequency]

#run DEICODE
qiime deicode rpca \
    --i-table ../../16S_Analyses/DEICODE/16S_directcomp/ESP_CTD_16S_master.biom.qza \
    --p-n-components 3 \
    --p-min-feature-count 10 \
    --p-min-sample-count 1000 \
    --o-biplot ../../16S_Analyses/DEICODE/16S_directcomp/ordination.qza \
    --o-distance-matrix ../../16S_Analyses/DEICODE/16S_directcomp/distance.qza
    
# Run PERMANOVA
# qiime diversity beta-group-significance \
#    --i-distance-matrix ../DEICODE/16S_directcomp/distance.qza \
#    --m-metadata-file ../DEICODE/16S_directcomp/directcomp_16S_sample_data_for_biom.txt \
#    --m-metadata-column sample_type.x \
#    --p-method permanova \
#    --o-visualization ../DEICODE/PERMANOVA_results/16S_ESP_CTD_PERMANOVA.qzv
```

### Plot DEICODE

```{r plot_COI_directcomp_PCA}
pco <- read_qza(here::here("16S_Analyses", "DEICODE", "16S_directcomp","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

## Prepare PCA data for ggplot

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))
# Subset metadata to only those samples that had > 1000 reads and were included in DEICODE
pca_metadata <- subset(yamahara_metadata_16S, sample_name %in% pca_data$SampleID)
pca_metadata %>% dplyr::rename(., "SampleID" = "sample_name") -> pca_metadata

# Join data from Kevan Yamahara's metadata
pca_metadata %>% 
  right_join(., pca_data, by = "SampleID") -> pca_data

# Use shapes for ESP vs. peristaltic pump
# Use colors for each of the different pairs
pair_colors <- c("1" = "#1f78b4" , "2" = "#33a02c", "3" = "#e31a1c", "4" = "#ff7f00", "5" = "#6a3d9a", "6" = "#b15928")
type_shapes <- c("combination_depth" = 21, "bench_ESP" = 24)


##----Run PERMANOVA-------------------------------------------------------------
# distance <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","16S_controls","distance.qza"))
# 
# # Extract distance matrix
# distance_matrix <- distance$data
# 
# # convert DEICODE matrix to "dist" class object
# PCA_dist <- as.dist(distance_matrix)
# 
# # PERMANOVA for control vs exp
# 
# permanova_control_envt_16S <- adonis2(PCA_dist ~ sample_type.x, data = pca_metadata, permutations=999)
# permanova_control_envt_16S_value <- permanova_control_envt_16S$`Pr(>F)`[1]

##----Make plot-----------------------------------------------------------------

directcomp_16S_PCA <- ggplot(pca_data,aes(x=PC1,y=PC2,color = sampling_pair, fill = sampling_pair, shape = Type))+
  geom_point(size = 5, stroke = 2)+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = pair_colors)+
  scale_fill_manual(values = pair_colors)+
  scale_shape_manual(values = type_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1))
  # ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)+ annotate(geom = "text", x = max(pca_data$PC1), y = max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/15, label = paste0("PERMANOVA (Environment vs. Control), p = ", permanova_control_envt_16S_value), hjust = 1, size = 5)

directcomp_16S_PCA

ggsave(path = fig_dir, filename = "directcomp_16S_PCA.png", directcomp_16S_PCA,width = 7,height = 5)
```




## DEICODE PCA - samples compared to CN18S and CN18F

### load new data

#### Load CN18S and CN18F data

```{r load_prep_combined_data}
# Read in combined taxonomy + ASV table
asv_tax_combined_path_16S <- here::here("data", "16S", "2017_2018",  "table_2017-2018_merged_filt_tax.tsv")
asv_tax_combined_16S <- read.table(asv_tax_combined_path_16S,  sep = "\t", header = TRUE)

# Split off asv table
dplyr::select(asv_tax_combined_16S, -Taxon) %>%
  column_to_rownames("ASV.ID") -> asv_table_16S

# Split off taxonomy, separate into fields per rank
tax_table_16S <- dplyr::select(asv_tax_combined_16S, ASV.ID, Taxon)
tax_table_16S$Taxon <- gsub("[a-zA-Z]__","", tax_table_16S$Taxon)
tax_table_16S %>% separate(., Taxon, into = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep = ";") %>% 
  column_to_rownames("ASV.ID") -> tax_table_16S
  
metadata_path_16S <- here::here("data", "16S", "2017_2018", "metadata_2017-2018.tsv")

ESP_CTD_16S_phyloseq <- merge_phyloseq(otu_table(asv_table_16S,taxa_are_rows = TRUE),
                                       tax_table(as.matrix(tax_table_16S)),
                                       sample_data(read.table(metadata_path_16S, row.names = 1, sep = "\t", header = TRUE)))

ESP_CTD_16S_directcomp_CN18FS_phyloseq <- prune_samples(sample_data(ESP_CTD_16S_phyloseq)$control_status != "control",ESP_CTD_16S_phyloseq)
ESP_CTD_16S_directcomp_CN18FS_phyloseq <- prune_taxa(taxa_sums(ESP_CTD_16S_directcomp_CN18FS_phyloseq) > 0, ESP_CTD_16S_directcomp_CN18FS_phyloseq)

# Remove the samples that were likely mislabeled
ESP_CTD_16S_directcomp_CN18FS_phyloseq <- prune_samples(!(sample_names(ESP_CTD_16S_directcomp_CN18FS_phyloseq) %in% c("CN17_B20", "CN17_CBI9", "CN17_B23", "CN17_CBI10")), ESP_CTD_16S_directcomp_CN18FS_phyloseq)

# Subset only the direct comparison samples
ESP_CTD_16S_directcomp_phyloseq <- prune_samples(sample_data(ESP_CTD_16S_directcomp_CN18FS_phyloseq)$sampling_cruise == "CN17S",ESP_CTD_16S_directcomp_CN18FS_phyloseq)

```


### check to make sure ASVs are shared
```{r}
# CN17S
ESP_CTD_16S_CN17S_phyloseq <- prune_samples(sample_data(ESP_CTD_16S_directcomp_CN18FS_phyloseq)$sampling_cruise == "CN17S",ESP_CTD_16S_directcomp_CN18FS_phyloseq)
ESP_CTD_16S_CN17S_phyloseq <- prune_taxa(taxa_sums(ESP_CTD_16S_CN17S_phyloseq) > 0, ESP_CTD_16S_CN17S_phyloseq)

# CN18S/CN18F
ESP_CTD_16S_CN18_phyloseq <- prune_samples(sample_data(ESP_CTD_16S_directcomp_CN18FS_phyloseq)$sampling_cruise %in% c("CN18F", "CN18S"), ESP_CTD_16S_directcomp_CN18FS_phyloseq)
ESP_CTD_16S_CN18_phyloseq <- prune_taxa(taxa_sums(ESP_CTD_16S_CN18_phyloseq) > 0, ESP_CTD_16S_CN18_phyloseq)

# extract ASV IDs
CN17S_ASVs <- unique(row.names(as.data.frame(tax_table(ESP_CTD_16S_CN17S_phyloseq))))
CN18_ASVs <- unique(row.names(as.data.frame(tax_table(ESP_CTD_16S_CN18_phyloseq))))

intersect(CN17S_ASVs, CN18_ASVs)
```




### Prep/export data for DEICODE
```{r prep_deicode_data_directcomp_CN18SF}
# Set directory for saving DEICODE data
deicode_dir <- here::here("16S_Analyses", "DEICODE", "16S_directcomp_CN18F_CN18S")

# Reformat and export OTU Table
directcomp_CN18FS_16S_asv_table <- as.data.frame(as(otu_table(ESP_CTD_16S_directcomp_CN18FS_phyloseq),"matrix"))
directcomp_CN18FS_16S_asv_table <- tibble::rownames_to_column(directcomp_CN18FS_16S_asv_table,"#OTUID")
write.table(directcomp_CN18FS_16S_asv_table, paste0(deicode_dir, "/directcomp_CN18FS_16S_asv_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_CN18FS_16S_tax_table <- as.data.frame(as(tax_table(ESP_CTD_16S_directcomp_CN18FS_phyloseq),"matrix"))
directcomp_CN18FS_16S_tax_table <- tibble::rownames_to_column(directcomp_CN18FS_16S_tax_table,"#OTUID")
write.table(directcomp_CN18FS_16S_tax_table,paste0(deicode_dir, "/directcomp_CN18FS_16S_tax_table_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
directcomp_CN18FS_16S_metadata <- as.data.frame(as.matrix(sample_data(ESP_CTD_16S_directcomp_CN18FS_phyloseq)))
directcomp_CN18FS_16S_metadata <- tibble::rownames_to_column(directcomp_CN18FS_16S_metadata,"#SampleID")
write.table(directcomp_CN18FS_16S_metadata, paste0(deicode_dir, "/directcomp_CN18FS_16S_sample_data_for_biom.txt"),sep = "\t",row.names = FALSE,quote = FALSE)
```


### Run DEICODE in Qiime2

```{bash engine.opts = '-l'}
# Enter qiime2 environment
conda activate qiime2-2020.11

#Make biom file
biom convert -i ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/directcomp_CN18FS_16S_asv_table_for_biom.txt -o ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/table.from_txt_json.biom --table-type="OTU table" --to-json
#add metadata files to biom file
biom add-metadata -i ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/table.from_txt_json.biom -o ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/table.w_md.biom \
--observation-metadata-fp ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/directcomp_CN18FS_16S_tax_table_for_biom.txt \
--sample-metadata-fp ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/directcomp_CN18FS_16S_sample_data_for_biom.txt

#import into Qiime2
qiime tools import \
 --input-path ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/table.w_md.biom \
 --output-path ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/ESP_CTD_16S_master.biom.qza \
 --type FeatureTable[Frequency]

#run DEICODE
qiime deicode rpca \
    --i-table ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/ESP_CTD_16S_master.biom.qza \
    --p-n-components 3 \
    --p-min-feature-count 10 \
    --p-min-sample-count 1000 \
    --o-biplot ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/ordination.qza \
    --o-distance-matrix ../../16S_Analyses/DEICODE/16S_directcomp_CN18F_CN18S/distance.qza
    
# Run PERMANOVA
# qiime diversity beta-group-significance \
#    --i-distance-matrix ../DEICODE/16S_directcomp/distance.qza \
#    --m-metadata-file ../DEICODE/16S_directcomp/directcomp_16S_sample_data_for_biom.txt \
#    --m-metadata-column sample_type.x \
#    --p-method permanova \
#    --o-visualization ../DEICODE/PERMANOVA_results/16S_ESP_CTD_PERMANOVA.qzv
```



### Plot DEICODE

```{r plot_16S_directcomp_PCA}
pco <- read_qza(here::here("16S_Analyses", "DEICODE", "16S_directcomp_CN18F_CN18S","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

## Prepare PCA data for ggplot

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))
# Subset metadata to only those samples that had > 1000 reads and were included in DEICODE
pca_metadata <- subset(yamahara_metadata_16S, sample_name %in% pca_data$SampleID)
pca_metadata %>% dplyr::rename(., "SampleID" = "sample_name") -> pca_metadata

# Join data from Kevan Yamahara's metadata
pca_metadata %>% 
  right_join(., pca_data, by = "SampleID") -> pca_data

# fill in pca_data with ESP/CTD
pca_data %>% 
  mutate(Type = ifelse(is.na(Type),"CTD/ESP",Type)) %>% 
  mutate(sampling_pair = ifelse(is.na(sampling_pair),"CTD/ESP", sampling_pair)) -> pca_data

# Use shapes for ESP vs. peristaltic pump
# Use colors for each of the different pairs
pair_colors <- c("1" = "#1f78b4" , "2" = "#33a02c", "3" = "#e31a1c", "4" = "#ff7f00", "5" = "#6a3d9a", "6" = "#b15928", "CTD/ESP" = "Gray50")
type_shapes <- c("combination_depth" = 21, "bench_ESP" = 24, "CTD/ESP" = 23)


##----Run PERMANOVA-------------------------------------------------------------
# distance <- read_qza(here::here("18S_12S_CO1_Analyses", "DEICODE","COI_controls","distance.qza"))
# 
# # Extract distance matrix
# distance_matrix <- distance$data
# 
# # convert DEICODE matrix to "dist" class object
# PCA_dist <- as.dist(distance_matrix)
# 
# # PERMANOVA for control vs exp
# 
# permanova_control_envt_COI <- adonis2(PCA_dist ~ sample_type.x, data = pca_metadata, permutations=999)
# permanova_control_envt_COI_value <- permanova_control_envt_COI$`Pr(>F)`[1]

##----Make plot-----------------------------------------------------------------

directcomp_16S_PCA <- ggplot(pca_data,aes(x=PC1,y=PC2,color = sampling_pair, fill = sampling_pair, shape = Type))+
  geom_point(size = 5, stroke = 2)+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = pair_colors)+
  scale_fill_manual(values = pair_colors)+
  scale_shape_manual(values = type_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1))
  # ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)+ annotate(geom = "text", x = max(pca_data$PC1), y = max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/15, label = paste0("PERMANOVA (Environment vs. Control), p = ", permanova_control_envt_16S_value), hjust = 1, size = 5)

directcomp_16S_PCA

# ggsave(path = fig_dir, filename = "directcomp_CN18SCN18F_PCA.png", directcomp_16S_PCA,width = 7,height = 5)
```




```{r}
pco <- read_qza(here::here("16S_Analyses", "DEICODE", "16S_directcomp_CN18F_CN18S","ordination.qza"))

label.PC1 <- paste0("PC1 (", round(pco$data$ProportionExplained$PC1, 3)*100,"%)")
label.PC1
label.PC2 <- paste0("PC2 (", round(pco$data$ProportionExplained$PC2, 3)*100,"%)")
label.PC2
label.PC3 <- paste0("PC3 (", round(pco$data$ProportionExplained$PC3, 3)*100,"%)")
label.PC3

pca_data <- pco$data$Vectors
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))

## Prepare PCA data for ggplot

pca_metadata <- directcomp_CN18FS_16S_metadata
pca_metadata %>% dplyr::rename(., "SampleID" = "#SampleID") -> pca_metadata
pca_data <- pco$data$Vectors
pca_data <- right_join(pca_data,pca_metadata,on = "SampleID")
pca_data <- subset(pca_data, !(is.na(pca_data$PC1)))

# Add a categorical depth field to the metadata
pca_data  %>%
  mutate(depth = as.numeric(depth_m)) %>%
  mutate(depth_cat = ifelse(depth <= 25.5, "0-25",
                            ifelse(depth <= 50, "27-50",
                                   ifelse(depth <= 201 & depth >=195, "195-200",
                                          "unknown")))) -> pca_data 

pca_data %>% 
  mutate(CTD_or_ESP = ifelse(depth_m == "mix", "dir. comp.", sampling_device)) %>% 
  mutate(CTD_or_ESP = ifelse(CTD_or_ESP == "CTD_Flyer", "CTD", CTD_or_ESP)) -> pca_data


# Shape guide for depths
# depth_shapes <- c("0-25" = 21, "27-50" = 22, "195-200" = 24)
cruise_colors <- c("CN18F" = "#ff7f00", "CN18S" = "#1f78b4", "CN17S" = "#e31a1c")
CTD_ESP_shapes <- c("CTD" = 21, "3G_ESP" = 24, "dir. comp." = 23)
# CTD_ESP_colors = c("CTD" = "#ff7f00", "ESP" = "#1f78b4")

##----Make plot-----------------------------------------------------------------

ESP_CTD_16S_PCA <- ggplot(pca_data, aes(x=PC1,y=PC2,color = sampling_cruise, fill = sampling_cruise, shape = CTD_or_ESP))+
  # Make fill for geom point be white if depth is deep; if depth is shallow, then fill with season
  geom_point(size = 5, stroke = 2, fill = ifelse(pca_data$depth_cat %in% c("195-200"), "white", 
                                     ifelse(pca_data$sampling_cruise == "CN18F", "#ff7f00", 
                                            ifelse(pca_data$sampling_cruise == "CN17S", "#e31a1c", "#1f78b4"))))+
  xlab(print(label.PC1))+
  ylab(print(label.PC2))+
  scale_color_manual(values = cruise_colors)+
  scale_fill_manual(values = cruise_colors)+
  scale_shape_manual(values = CTD_ESP_shapes)+
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 20),
        axis.text = element_blank(),
        axis.ticks.length=unit(0.25, "cm"),
        axis.ticks=element_blank(),
        plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
        axis.line.x.bottom = element_line(color = "black", size = 1),
        axis.line.y.left = element_line(color = "black", size = 1),
        legend.position = "none")+
  ylim(min(pca_data$PC2)-abs(max(pca_data$PC2)-min(pca_data$PC2))/30, max(pca_data$PC2)+abs(max(pca_data$PC2)-min(pca_data$PC2))/12)

ESP_CTD_16S_PCA

ggsave(path = fig_dir, filename = "directcomp_16S_CN18SCN18F_PCA.png", ESP_CTD_16S_PCA,width = 7,height = 5)
```

## Compare taxa

### Plot direct comparison taxa - top 20 ASVs

#### Make a function for plotting
```{r plot_top_20ASVs_function}
fig_dir <- here("figures", "direct_comparisons", "16S")

plot_top20ASVs_directcomp <- function(sample_names, plot_name){
  # Subset the samples specified in the function
  phyloseq <- prune_samples(sample_names(directcomp_16S_phyloseq) %in% sample_names, directcomp_16S_phyloseq)
  
  directcomp_16S_asv_table <- as.data.frame(otu_table(phyloseq))
directcomp_16S_asv_table <- tibble::rownames_to_column(directcomp_16S_asv_table,"ASV")

directcomp_16S_tax_table <- as.data.frame(as(tax_table(phyloseq),"matrix"))
directcomp_16S_tax_table <- tibble::rownames_to_column(directcomp_16S_tax_table,"ASV")
directcomp_16S_tax_table %>% mutate_all(as.character) -> directcomp_16S_tax_table

# Rename ASVs with their taxonomy
directcomp_16S_asv_table <- left_join(directcomp_16S_asv_table, directcomp_16S_tax_table, by = "ASV")

directcomp_16S_asv_table %>% mutate(.,ASV_name = ifelse(!(Species %in% c("s_","unassigned", "")),paste0(ASV,"_",Species),
                                                    ifelse(!(Genus %in% c("g_","unassigned", "")),paste0(ASV,"_",Genus),
                                                           ifelse(!(Family %in% c("unassigned", "")),paste0(ASV,"_",Family),
                                                                  ifelse(!(Order %in% c("unassigned", "")),paste0(ASV,"_",Order),
                                                                         ifelse(!(Class %in% c("unassigned", "")),paste0(ASV,"_",Class),
                                                                                ifelse(!(Phylum %in% c("unassigned", "")),paste0(ASV,"_",Phylum),
                                                                                       paste0(ASV,"_","unassigned")))))))) -> directcomp_16S_asv_table
# Convert to long form
gathercols <- colnames(as.data.frame(otu_table(phyloseq)))

directcomp_16S_asv_table_gather <- gather(directcomp_16S_asv_table, key = "sample", value = "reads", gathercols, factor_key=TRUE)


# Figure out the top 20 ASVs total in the dataset to plot
directcomp_16S_asv_table_asv_total_reads <- as.data.frame(sort(-taxa_sums(phyloseq)))
top_19_CN18S_blanks_asvs <- rownames(directcomp_16S_asv_table_asv_total_reads)[1:19]

directcomp_16S_asv_table_gather %>% mutate(.,ASV_name_updated = ifelse(ASV %in% top_19_CN18S_blanks_asvs,ASV_name,"other")) %>% 
  mutate(sample = as.character(sample))-> directcomp_16S_asv_table_gather_updated

# Add metadata
yamahara_metadata_16S %>% 
  dplyr::rename(sample = sample_name) %>%
  right_join(., directcomp_16S_asv_table_gather_updated, by = "sample") %>% 
  mutate(Type = ifelse(Type == "combination_depth", "peristaltic_pump", "benchtop_ESP")) %>% 
  mutate(pair = paste0(sampling_pair, "_", Type)) -> directcomp_16S_asv_table_gather_updated

directcomp_16S_gg <- ggplot(directcomp_16S_asv_table_gather_updated,aes(x = pair, y = reads, fill = ASV_name_updated))+
  geom_bar(stat = "identity")+
  ggtitle(plot_name)+
  scale_fill_tableau("Tableau 20")+
  # ylim(0,20000)+
  theme(legend.text = element_text(size = 7),
        axis.text.x = element_text(angle = 90),
        axis.title.x = element_blank())+
  guides(fill = guide_legend(title = "ASV"))

# directcomp_16S_gg
# return(directcomp_16S_gg)
ggsave(paste0(fig_dir, "/", plot_name, ".png"), directcomp_16S_gg, height = 6.5, width = 8)
}

```


#### All samples together
```{r}
directcomp_16S_asv_table <- as.data.frame(otu_table(directcomp_16S_phyloseq))
directcomp_16S_asv_table <- tibble::rownames_to_column(directcomp_16S_asv_table,"ASV")

directcomp_16S_tax_table <- as.data.frame(as(tax_table(directcomp_16S_phyloseq),"matrix"))
directcomp_16S_tax_table <- tibble::rownames_to_column(directcomp_16S_tax_table,"ASV")
directcomp_16S_tax_table %>% mutate_all(as.character) -> directcomp_16S_tax_table

# Rename ASVs with their taxonomy
directcomp_16S_asv_table <- left_join(directcomp_16S_asv_table, directcomp_16S_tax_table, by = "ASV")

directcomp_16S_asv_table %>% mutate(.,ASV_name = ifelse(!(Species %in% c("s_","unassigned")),paste0(ASV,"_",Species),
                                                    ifelse(!(Genus %in% c("g_","unassigned")),paste0(ASV,"_",Genus),
                                                           ifelse(!(Family == "unassigned"),paste0(ASV,"_",Family),
                                                                  ifelse(!(Order == "unassigned"),paste0(ASV,"_",Order),
                                                                         ifelse(!(Class == "unassigned"),paste0(ASV,"_",Class),
                                                                                ifelse(!(Phylum == "unassigned"),paste0(ASV,"_",Phylum),
                                                                                       paste0(ASV,"_","unassigned")))))))) -> directcomp_16S_asv_table
# Convert to long form
gathercols <- colnames(as.data.frame(otu_table(directcomp_16S_phyloseq)))

directcomp_16S_asv_table_gather <- gather(directcomp_16S_asv_table, key = "sample", value = "reads", gathercols, factor_key=TRUE)


# Figure out the top 20 ASVs total in the dataset to plot
directcomp_16S_asv_table_asv_total_reads <- as.data.frame(sort(-taxa_sums(directcomp_16S_phyloseq)))
top_19_CN18S_blanks_asvs <- rownames(directcomp_16S_asv_table_asv_total_reads)[1:19]

directcomp_16S_asv_table_gather %>% mutate(.,ASV_name_updated = ifelse(ASV %in% top_19_CN18S_blanks_asvs,ASV_name,"other")) -> directcomp_16S_asv_table_gather_updated

# Add metadata
directcomp_meta_yamahara %>% 
  dplyr::rename(sample = SampleID) %>% 
  right_join(., directcomp_16S_asv_table_gather_updated, by = "sample") %>% 
  mutate(Type = ifelse(Type == "combination_depth", "peristaltic_pump", "benchtop_ESP")) %>% 
  mutate(pair = paste0(sampling_pair, "_", Type)) -> directcomp_16S_asv_table_gather_updated

directcomp_16S_gg <- ggplot(directcomp_16S_asv_table_gather_updated,aes(x = pair, y = reads, fill = ASV_name_updated))+
  geom_bar(stat = "identity")+
  ggtitle("Direct comparisons")+
  scale_fill_tableau("Tableau 20")+
  # ylim(0,20000)+
  theme(legend.text = element_text(size = 7),
        axis.text.x = element_text(angle = 90),
        axis.title.x = element_blank())+
  guides(fill = guide_legend(title = "ASV"))

directcomp_16S_gg

ggsave(paste0(fig_dir, "/directcomp_16S_top20ASVs.png"), directcomp_16S_gg, height = 6.5, width = 8)
```

#### Plot individual sample pairs
```{r plot_sample_pairs}
plot_top20ASVs_directcomp(sample_names = subset(yamahara_metadata_16S, sampling_pair == "1")$sample_name, plot_name = "directcomp_16S_pair1")

plot_top20ASVs_directcomp(sample_names = subset(yamahara_metadata_16S, sampling_pair == "2")$sample_name, plot_name = "directcomp_16S_pair2")

plot_top20ASVs_directcomp(sample_names = subset(yamahara_metadata_16S, sampling_pair == "3")$sample_name, plot_name = "directcomp_16S_pair3")

plot_top20ASVs_directcomp(sample_names = subset(yamahara_metadata_16S, sampling_pair == "4")$sample_name, plot_name = "directcomp_16S_pair4")

plot_top20ASVs_directcomp(sample_names = subset(yamahara_metadata_16S, sampling_pair == "5")$sample_name, plot_name = "directcomp_16S_pair5")

plot_top20ASVs_directcomp(sample_names = subset(yamahara_metadata_16S, sampling_pair == "6")$sample_name, plot_name = "directcomp_16S_pair6")
```


# Make combined direct comparison figure - phyla for 16S + COI

## Reformat 16S data
```{r prep_data_16S}
# Extract ASVs
as.data.frame(otu_table(ESP_CTD_16S_directcomp_phyloseq)) %>% 
  rownames_to_column("ASV") -> directcomp_16S_ASV_table

# Extract taxonomy table
directcomp_16S_tax_table <- as.data.frame(as(tax_table(ESP_CTD_16S_directcomp_phyloseq),"matrix"))
directcomp_16S_tax_table <- tibble::rownames_to_column(directcomp_16S_tax_table,"ASV")

# reformat ASV table as long form
directcomp_16S_ASV_table %>% 
  pivot_longer(cols = colnames(directcomp_16S_ASV_table)[2:ncol(directcomp_16S_ASV_table)], names_to = "sample_name") -> directcomp_16S_asv_table_long

directcomp_16S_asv_table_long %>% 
  left_join(directcomp_16S_tax_table, by = "ASV") -> DC_16S_table

# Add shortened sample names
directcomp_meta_yamahara %>% 
  mutate(name = paste0(sampling_pair, ": ", ifelse(Type == "combination_depth", "peristaltic \npump     ", 
                                                   ifelse(Type == "bench_ESP", "benchtop \nESP     ",Type)))) %>% 
  mutate(sample_name = gsub("CANON", "CN17", sample_name)) %>% 
  mutate(sample_name = gsub("_22", "", sample_name)) -> directcomp_meta_yamahara_for_16S_join

DC_16S_table %>% 
  left_join(., dplyr::select(directcomp_meta_yamahara_for_16S_join, sample_name, name, Type, sampling_pair), by = "sample_name") -> DC_16S_table

# Keep only top four pairs (due to likely sample mislabeling with pairs 5 and 6)
DC_16S_table <- subset(DC_16S_table, sampling_pair %in% c("1", "2", "3", "4"))

# unique(DC_16S_table$Phylum)
```

```{r determine_top_phyla_16S}
DC_16S_table %>% 
  # Change so that "unassigned", "no_hit", and "unknown" will merge together
  mutate(Phylum = gsub(" ", "", Phylum)) %>% 
  mutate(., Phylum = ifelse(Phylum %in% c(""), "Unassigned", Phylum)) %>% 
  group_by(Phylum) %>% 
  summarise(total = sum(value)) -> DC_16S_phyla_reads


DC_16S_phyla_reads %>% 
  arrange(., desc(total)) -> DC_16S_phyla_reads

# Get the top ten phyla; we need top ten, then "unknown" and "other" to get to 12
# Get top 11
top11_phyla_16S <- DC_16S_phyla_reads$Phylum[1:11]

# Mutate data for plotting to get new phyla (top 10 + unassigned + other)
DC_16S_table %>% 
  mutate(Phylum = gsub(" ", "", Phylum)) %>% 
  mutate(., Phylum = ifelse(Phylum %in% c(""), "Unassigned", Phylum)) %>% 
  mutate(., Phylum = ifelse(Phylum %in% c("unassigned", "no_hit", "unknown"), "Unassigned", Phylum)) %>% 
  mutate(., phylum2 = ifelse(Phylum %in% top11_phyla_16S, Phylum, "Other")) -> DC_16S_table

# Calculate proportion of total
DC_16S_table %>% 
  group_by(name) %>% 
  mutate(rel_reads = value/sum(value)*100) -> DC_16S_table

# Group by phylum and collapse table
DC_16S_table %>% 
  group_by(name, phylum2) %>% 
  summarise(rel_reads = sum(rel_reads)) -> DC_16S_table_forplot

# Rename two phyla to shorten names
DC_16S_table_forplot %>% 
  mutate(phylum2 = ifelse(phylum2 == "SAR324_clade(Marine_group_B)", "SAR324 Clade", 
                          ifelse(phylum2 == "Marinimicrobia_(SAR406_clade)", "Marinimicrobia", phylum2))) -> DC_16S_table_forplot

# Reorder phyla so that unassigned and other are at the bottom
DC_16S_table_forplot$phylum2 <- factor(DC_16S_table_forplot$phylum2, levels = c("Actinobacteriota", "Bacteroidota", "Crenarchaeota",                "Marinimicrobia", "Nitrospinota",  "Planctomycetota", "Proteobacteria", "SAR324 Clade",  "Thermoplasmatota",  "Verrucomicrobiota",  "Other", "Unassigned"))

# Reorder samples
DC_16S_table_forplot$name <- factor(DC_16S_table_forplot$name, levels = c("4: benchtop \nESP     ", "4: peristaltic \npump     ",
                                                                          "3: benchtop \nESP     ", "3: peristaltic \npump     ",
                                                                          "2: benchtop \nESP     ", "2: peristaltic \npump     ",
                                                                          "1: benchtop \nESP     ", "1: peristaltic \npump     "))

# Make color scale
phylum_colors <- c("#a6cee3",
"#1f78b4",
"#b2df8a",
"#33a02c",
'#fb9a99',
"#e31a1c",
"#fdbf6f",
"#ff7f00",
"#cab2d6",
"#6a3d9a",
"#b15928",
"gray50")
# "#ffff99",
# "#b15928")
```

```{r plot_top_phyla}
DC_16S_top_phyla_plot <- ggplot(DC_16S_table_forplot, aes(x = name, y = rel_reads, fill = phylum2)) +
  geom_bar(stat = "identity", color = "black", size = 0.3) +
  scale_fill_manual(values = phylum_colors)+
  labs(x = "Sample", y = "Percentage of total reads") +
  scale_y_reverse(expand = c(0,0), labels = c("100", "75", "50", "25", "0")) +
  coord_flip()+
  scale_x_discrete(expand = c(0,0)) + 
  guides(fill = guide_legend(title = "Phylum", nrow = 4, byrow = TRUE, title.position = "top")) +
  theme(panel.background = element_rect(fill = "white",size = 1),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        axis.title = element_text(size = 14),
        # axis.text = element_blank(),
        axis.ticks.length.y=unit(0, "cm"),
        axis.ticks.length.x = unit(0.1, "cm"),
        axis.text.y = element_text(size = 12),
        # axis.ticks=element_blank(),
        plot.margin = margin(0.5, 0.25, 0.25, 0.25, "cm"),
        legend.position = "top",
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 15),
        legend.title.align = 0.5)
        # axis.line.x.bottom = element_line(color = "black", size = 1),
        # axis.line.y.left = element_line(color = "black", size = 1))

DC_16S_top_phyla_plot

ggsave(paste0(fig_dir, "/16S_directcomp_phyla_plot.png"), DC_16S_top_phyla_plot, height = 8, width = 6)



# Second version of plot with shapes as y axis labels
# 
# # Add sampling pair and filtration method to table
# DC_shapes_y_axis = data.frame(name = c("1: peristaltic pump", "1: benchtop ESP", 
#                                        "2: peristaltic pump", "2: benchtop ESP", 
#                                        "3: peristaltic pump", "3: benchtop ESP", 
#                                        "4: peristaltic pump", "4: benchtop ESP"),
#                               filtration_method = c("benchtop ESP", "peristaltic pump",
#                                        "benchtop ESP", "peristaltic pump",
#                                        "benchtop ESP", "peristaltic pump",
#                                        "benchtop ESP", "peristaltic pump"),
#                               sampling_pair = as.character(c(1,1,2,2,3,3,4,4)),
#                               # pos.x = c(-1,-1,-1,-1,-1,-1,-1,-1),
#                               pos.x = rep(105,8),
#                               pos.y = c(8,7,6,5,4,3,2,1))
# 
# filt_method_shapes <- c("benchtop ESP" = 17,	"peristaltic pump" = 19)
# 
# DC_16S_top_phyla_plot <- ggplot(DC_16S_table_forplot, aes(x = name, y = rel_reads, fill = phylum2)) +
#   geom_bar(stat = "identity", color = "black", size = 0.3) +
#   scale_fill_manual(values = phylum_colors)+
#   labs(x = "Sample", y = "Percentage of total reads") +
#   scale_y_reverse(expand = c(0,0), labels = c("100", "75", "50", "25", "0"), limits = c(110, -5)) +
#   coord_flip()+
#   scale_x_discrete(expand = c(0,0)) + 
#   guides(fill = guide_legend(title = "Phylum", nrow = 3, byrow = TRUE, title.position = "top")) +
#   theme(panel.background = element_rect(fill = "white",size = 1),
#         panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
#         axis.title = element_text(size = 14),
#         # axis.text = element_blank(),
#         axis.ticks.length.y=unit(0, "cm"),
#         axis.ticks.length.x = unit(0.1, "cm"),
#         axis.text.y = element_blank(),
#         axis.text.x = element_text(size = 12),
#         # axis.ticks=element_blank(),
#         # plot.margin = margin(0.25, 0.25, 0.25, 0.25, "cm"),
#         legend.text = element_text(size = 12),
#         legend.title = element_text(size = 15),
#         legend.position = "top",
#         legend.title.align = 0.5)+
#   # Add the points 
#   geom_point(data = DC_shapes_y_axis, inherit.aes = FALSE, aes(x = pos.y, y = pos.x, shape = filtration_method, color = sampling_pair), size = 6) +
#     scale_color_manual(values = pair_colors)+
#   scale_shape_manual(values = filt_method_shapes)+
#   guides(shape = FALSE, color = FALSE)
#         # axis.line.x.bottom = element_line(color = "black", size = 1),
#         # axis.line.y.left = element_line(color = "black", size = 1))
# 
# DC_16S_top_phyla_plot

```

### Join 16S and COI top phyla plots using ggarrange
```{r make_directcomp_fig}
DC_COI_16S_phylum_fig <- ggarrange(DC_16S_top_phyla_plot, DC_COI_top_phyla_plot, heights = c(7,7), widths = c(5,4.1), labels = c("(a) 16S", "(b) COI"),
                                   label.x = 0.01, label.y = 0.995, font.label = list(size = 24), hjust = 0)

ggsave(paste0(fig_dir, "/final_figures/COI_16S_directcomp_phylum_fig.png"), DC_COI_16S_phylum_fig, height = 8, width = 12)
```




